macros/generate_base_model.sql

```diff
@@ -1,10 +1,13 @@
-{% macro generate_base_model(source_name, table_name, leading_commas=False, case_sensitive_cols=False) %}
+{% macro generate_base_model(source_name, table_name, leading_commas=False, case_sensitive_cols=False, materialization='table') %}
 
 {%- set source_relation = source(source_name, table_name) -%}
 
 {%- set columns = adapter.get_columns_in_relation(source_relation) -%}
 {% set column_names=columns | map(attribute='name') %}
 {% set base_model_sql %}
+
+{{ "{{ config(materialized='"+materialization+"') }}" }}
```

Comment:
This will need a block that checks whether `materialized` arg is `None`. 

Also I have a slight preference for `~` for concatenation instead of `+`, and a bit of whitespace wouldn't go astray
```suggestion
{{ "{{ config(materialized='" ~ materialization ~ "') }}" }}
```

Reply:
Done, cheers!